src/Microsoft.HttpRepl.Tests/OpenApi/EndpointMetadataReaderTests.cs

```diff
@@ -0,0 +1,495 @@
+        }
+
+        [Fact]
+        public void Read_WithSwaggerV2InputAndNoPaths_ReturnsEmptyListOfEndPointMetaData()
+        {
+            string json = @"{
+  ""swagger"": ""2.0"",
+  ""info"": {
+    ""version"": ""v1""
+  }
+}";
+            List<EndpointMetadata> endpointMetadata = GetEndpointMetadataList(json);
+
+            Assert.Empty(endpointMetadata);
+        }
+
+        [Fact]
+        public void Read_WithSwaggerV2InputAndNoProperties_ReturnsEmptyListOfEndPointMetaData()
+        {
+            string json = @"{
+  ""swagger"": ""2.0"",
+  ""info"": {
+    ""version"": ""v1""
+  },
+  ""paths"": {
+  }
+}";
+            List<EndpointMetadata> endpointMetadata = GetEndpointMetadataList(json);
+
+            Assert.Empty(endpointMetadata);
+        }
+
+        [Fact]
+        public void Read_WithSwaggerV2InputAndNoRequestMethods_ReturnsEndpointMetadataWithEmptyAvailableRequests()
+        {
+            string json = @"{
+  ""swagger"": ""2.0"",
+  ""info"": {
+    ""version"": ""v1""
+  },
+  ""paths"": {
+    ""/api/Employees"": {
+    }
+  }
+}";
+            List<EndpointMetadata> endpointMetadata = GetEndpointMetadataList(json);
+            IReadOnlyDictionary<string, IReadOnlyDictionary<string, IReadOnlyList<Parameter>>> availableRequests = endpointMetadata[0].AvailableRequests;
+
+            Assert.Single(endpointMetadata);
+            Assert.Equal("/api/Employees", endpointMetadata[0].Path);
+            Assert.Empty(availableRequests);
+        }
+
+        [Fact]
+        public void Read_WithSwaggerV2Input_ReturnsEndpointMetadata()
+        {
+            string json = @"{
+  ""swagger"": ""2.0"",
+  ""info"": {
+    ""version"": ""v1""
+  },
+  ""paths"": {
+    ""/api/Employees"": {
+      ""get"": {
+        ""tags"": [
+          ""Employees""
+        ],
+        ""operationId"": ""GetEmployee"",
+        ""consumes"": [],
+        ""produces"": [
+          ""text/plain""
+        ],
+        ""parameters"": [],
+        ""responses"": {
+          ""200"": {
+            ""description"": ""Success""
+          }
+        }
+      },
+      ""post"": {
+        ""tags"": [
+          ""Employees""
+        ],
+        ""operationId"": ""put"",
+        ""consumes"": [],
+        ""produces"": [
+          ""text/plain""
+        ],
+        ""parameters"": [
+          {
+            ""name"": ""id"",
+            ""in"": ""path""
+          }
+        ],
+        ""responses"": {
+          ""200"": {
+            ""description"": ""Success""
+          }
+        }
+      }
+    }
+  }
+}";
+            List<EndpointMetadata> endpointMetadata = GetEndpointMetadataList(json);
+            IReadOnlyDictionary<string, IReadOnlyDictionary<string, IReadOnlyList<Parameter>>> availableRequests = endpointMetadata[0].AvailableRequests;
+            KeyValuePair<string, IReadOnlyDictionary<string, IReadOnlyList<Parameter>>> firstRequest = availableRequests.First();
```

Comment:
Is the enumeration order of the dictionary guaranteed to be consistent?  I.e. will you always get the same results for `.First()` and `.ElementAt(1)`?

Reply:
In practice, probably, but the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2) says it is non-deterministic.

> For purposes of enumeration, each item in the dictionary is treated as a KeyValuePair<TKey,TValue> structure representing a value and its key. The order in which the items are returned is undefined.

Reply:
Updated to verify dictionary contains key instead:
https://github.com/aspnet/HttpRepl/commit/35818741ff04b0665aacfffe40103dbc9635a00b